### PR TITLE
feat(webhooks/github): Add more metadata to PR events

### PR DESCRIPTION
**Pull request by jervi**
Add PR number, draft status, state and title to the pull request event.
Also did some cleanup of the `GithubWebhookEvent` interface and removed unused parameters.

### CODE REVIEW

File: echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandler.java [hunk 1]
```diff
@@ -79,14 +76,22 @@ public void handle(Event event, Map postedEvent) {
...
-    results.put(
-        "action", githubEvent.concat(":").concat(webhookEvent.getAction(event, postedEvent)));
+    results.put("repoProject", webhookEvent.getRepoProject());
+    results.put("slug", webhookEvent.getSlug());
+    results.put("hash", webhookEvent.getHash());
+    results.put("branch", webhookEvent.getBranch());
+    results.put("action", githubEvent.concat(":").concat(webhookEvent.getAction()));
+    if (webhookEvent instanceof GithubPullRequestEvent) {
+      GithubPullRequestEvent pullRequestEvent = (GithubPullRequestEvent) webhookEvent;
+      if (((GithubPullRequestEvent) webhookEvent).getPullRequest() != null) {
```
Comment (mattgogerly) [inline echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandler.java:88]: Use `pullRequestEvent` assigned above to remove the need to cast?

File: echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/github/GithubPullRequestEvent.java [hunk 2]
```diff
@@ -17,23 +17,21 @@
...
   @JsonProperty("pull_request")
-  PullRequest pullRequest;
+  private PullRequest pullRequest;
 
-  String action;
+  private String action;
 
   // `.repository.full_name`
   @Override
-  public String getFullRepoName(Event event, Map postedEvent) {
```
Comment (mattgogerly) [inline echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/github/GithubPullRequestEvent.java:36]: Are these args not actually used in any of these methods?

File: echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandler.java (hunk 1 above)
Comment (jervi) [reply echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandler.java:88]: omg, yes :p

File: echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/github/GithubPullRequestEvent.java (hunk 2 above)
Comment (jervi) [reply echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/github/GithubPullRequestEvent.java:36]: Nope, nowhere.

File: echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandler.java [hunk 3]
```diff
@@ -79,14 +76,22 @@ public void handle(Event event, Map postedEvent) {
...
-    results.put("hash", webhookEvent.getHash(event, postedEvent));
-    results.put("branch", webhookEvent.getBranch(event, postedEvent));
-    results.put(
-        "action", githubEvent.concat(":").concat(webhookEvent.getAction(event, postedEvent)));
+    results.put("repoProject", webhookEvent.getRepoProject());
+    results.put("slug", webhookEvent.getSlug());
+    results.put("hash", webhookEvent.getHash());
+    results.put("branch", webhookEvent.getBranch());
+    results.put("action", githubEvent.concat(":").concat(webhookEvent.getAction()));
+    if (webhookEvent instanceof GithubPullRequestEvent) {
```
[the remark this answers is not in the document]
Comment (jervi) [reply echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandler.java:86]: Will fix

**Review by mattgogerly — APPROVED** [review]

**Comment by jervi** [comment]
Wait a bit with merging this - I need to do some additional changes...

**Comment by jervi** [comment]
Finally it works. I put the extra properties on the `properties` element to avoid adding even more fields to the `Trigger` class. An example:

```json
"trigger": {
    "type": "git",
    "user": "anonymous",
    "parameters": {},
    "artifacts": [],
    "notifications": [],
    "rebake": false,
    "dryRun": false,
    "strategy": false,
    "hash": "a37cfa65879fa21a6c03738680de888be77855eb",
    "source": "github",
    "project": "myorg",
    "branch": "my-branch",
    "slug": "jervi-is-testing-echo",
    "action": "pull_request:reopened",
    "resolvedExpectedArtifacts": [],
    "expectedArtifacts": [],
    "eventId": "cb3531f1-54e0-4e04-952a-e39c4408ee60",
    "secret": "mysecret",
    "enabled": true,
    "id": "320732db-9c80-30bf-96d8-c17383e953ff",
    "preferred": false,
    "properties": {
        "number": 8,
        "draft": false,
        "state": "open",
        "title": "Add README.md"
    }
}
```

**Pull request merged**